src/core/sprites/webgl/SpriteRenderer.js

```diff
@@ -108,16 +109,13 @@ export default class SpriteRenderer extends ObjectRenderer
         // step 2: check the maximum number of if statements the shader can have too..
         this.MAX_TEXTURES = checkMaxIfStatmentsInShader(this.MAX_TEXTURES, gl);
 
-        this.shaders = new Array(this.MAX_TEXTURES);
-        this.shaders[0] = generateMultiTextureShader(gl, 1);
-        this.shaders[1] = generateMultiTextureShader(gl, 2);
+        const shader = this.shader = generateMultiTextureShader(gl, this.MAX_TEXTURES);
```

Comment:
In all my tests, this does not seem to make a difference (which is great!) The main reason I set up the shaders like this was to mitigate a webGL error that was introduced about half a year ago where it complained about using textures that are not bound. Fortunately this restructure solves it in a different way.

For some mobile devices we still set max textures to 4 as even though they claim to be able to have more that, performance begins to drop because of the branching. 

Correct me if I'm wrong here too. But I believe that the style of these if statements work pretty fast as they is good Cache coherence.